pandas/core/base.py

```diff
@@ -356,7 +356,11 @@ def _aggregate(self, arg, *args, **kwargs):
                 if isinstance(obj, ABCDataFrame) and len(
                     obj.columns.intersection(keys)
                 ) != len(keys):
-                    raise SpecificationError("nested renamer is not supported")
+                    cols = sorted(set(keys) - set(obj.columns.intersection(keys)))
```

Comment:
its not worth it dispatching on both, just make. single message

Reply:
Fixed both of your suggestions - am getting a lot of seemingly unrelated failures on pandas/tests/groupby/aggregate/test_other.py though. Should I push anyway?